onboarding/source/controller.c

```diff
@@ -33,68 +33,91 @@ static void lightTimerCallback(TimerHandle_t xTimer);
  */
 static void ledTimerCallback(TimerHandle_t xTimer);
 
-uint8_t initController(void) {
+uint8_t initController(void)
+{
     BaseType_t xReturned = pdFAIL;
-    if (controllerTaskHandle == NULL) {
+    if (controllerTaskHandle == NULL)
+    {
         // Create controller task
-        xReturned = xTaskCreate(controllerTask,             /* Function that implements the task. */
-                                CONTROLLER_NAME,            /* Text name for the task. */
-                                CONTROLLER_STACK_SIZE,      /* Stack size in words, not bytes. */
-                                NULL,                       /* Parameter passed into the task. */
-                                CONTROLLER_PRIORITY,        /* Priority at which the task is created. */
-                                &controllerTaskHandle);     /* Used to pass out the created task's handle. */
+        xReturned = xTaskCreate(controllerTask,         /* Function that implements the task. */
+                                CONTROLLER_NAME,        /* Text name for the task. */
+                                CONTROLLER_STACK_SIZE,  /* Stack size in words, not bytes. */
+                                NULL,                   /* Parameter passed into the task. */
+                                CONTROLLER_PRIORITY,    /* Priority at which the task is created. */
+                                &controllerTaskHandle); /* Used to pass out the created task's handle. */
     }
 
```

Comment:
If we failed to create the task, there's no point in creating the timers. Check xReturned before creating the timers.